### DISCUSSION TITLE
authentication before accepting a connection

### DISCUSSION THREAD

**Discussion by garyn87048** [micropython/micropython#12310] [Ideas]
I would like mpremote to have an authentication feature before accepting a connection.  

I have esp32s on my network doing specific tasks and I prefer to not have their tasks mistakenly interrupted because of an inadvertent connection.  For example, to connect to the temperature monitoring esp32 would required $ mpremote connect addr --user thermo --pass onewire and to connect to the zone pump esp32 would require something like --user zones --pass pumpcontrol.  At the moment, mpremote connect attaches to the first esp it can find.

Apologies if I missed and existing method for controlling access!

Thanks!

**Comment by peterhinch** [comment]
There is no such mechanism. As far as I know you are the first person to request this so I rather doubt that it will happen. The usual solution is to unplug "do not tamper" devices from the PC, powering them from another source.

**Comment by Josverl** [comment]
Mpremote over telnet could do this, the main issue is the lack of a good (fast, stable, small) telnet server to connect to

Another option would be rfc2217 serial over TCP support. I have a mpremote version with support for this, and the protocol supports basic authentication similar to telnet

However the pyserial Rfc2217 server does NOT provide security according to the docs. 
> NOTE: no security measures are implemented. Anyone can remotely connect to this service over the network.

 
So I guess the main question is if you need to secure the serial or the network access, and if there is a device that could run a telnet or secure rfc2217 server

Ref:
https://docs.espressif.com/projects/esptool/en/latest/esp8266/esptool/remote-serial-ports.html#pyserial-example-servers

**Comment by garyn87048** [comment]
Thanks!  I got both devices disconnected from USB power and remotely accessed with webrepl.  That solved the problem of inadvertently stopping them by connecting with mpremote.  My esp32 noviceness!
